pandas/_libs/lib.pyx

```diff
@@ -2204,6 +2205,8 @@ def maybe_convert_numeric(
     bint convert_empty=True,
     bint coerce_numeric=False,
     bint convert_to_masked_nullable=False,
+    str thousands=None,
```

Comment:
Do these need to be `str` in the first place or would a declaration of `char` simplify things? I am not sure how Cython works in this case but it seems like it might be able to automatically handle that object -> ctype conversion

Reply:
This was my original implementation, and it ran fine, but I ran into issues with the mypy tests. It wasn't matching the stub function to its implentation here as the parameter types were different: I couldn't use char dtype in the .pyi, and I was trying to declare it as bytes. It may work to import the cython dtypes into the .pyi file but I'm not sure if this is the best option or there is another neat method.